react-express-mongodb/server/server.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Created by Syed Afzal
+ */
+require('./config/config');
+
+const express = require('express');
+const path = require('path');
+const cookieParser = require('cookie-parser');
+const bodyParser = require('body-parser');
+const cors = require('cors');
+const db = require('./db');
+
+
+
+const app = express();
+
+//connection from db here
+db.connect(app);
+
+app.use(cors());
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname,    'public')));
+
+//  adding routes
+require('./routes')(app);
+
+app.listen(3000, () => {
+    console.log("Server is up on port", 3000)
+});
```

Comment:
if we replace by the following code, the app will wait for the connection with mongo before exposing the http server 😉
```
app.on('ready', () => {
    app.listen(3000, () => {
        console.log("Server is up on port", 3000)
    });
})
```

Reply:
Done